libsel4vmmplatsupport/src/drivers/virtio_console_emul.c

```diff
@@ -33,14 +37,14 @@ static void emul_con_rx_complete(virtio_emul_t *emul, int queue, char *buf, unsi
     uint16_t idx = virtq->last_idx[queue];
 
     if (idx != guest_idx) {
-        /* total length of the written packet so far */
-        size_t tot_written = 0;
         /* amount of the current descriptor written */
         size_t desc_written = 0;
         /* how much we have written of the current buffer */
         size_t buf_written = 0;
-        /* the current buffer. -1 indicates the virtio con buffer */
-        int current_buf = 0;
+        /* len of the data that we need to write */
+        size_t len = (tail > head) ? (tail - head) : (tail - head + VUART_BUFLEN);
```

Comment:
Is VUART_BUFLEN the same as BUFSIZE in the last PR?

Reply:
Yes.